Dockerfile

```
@@ -0,0 +1,28 @@
+# [1] Build Stage
+FROM gradle:8.5.0-jdk17-alpine AS build
+WORKDIR /app
+# 변경 적은 것 먼저 복사해서 캐시 활용
+COPY build.gradle ./
+COPY settings.gradle ./
+RUN gradle dependencies --no-daemon || true
+
+# 변경 자주 되는 것 아래로 ↓
+COPY . .
+
+# 빌드 실행 (테스트 생략)
+RUN chmod +x gradlew
+RUN ./gradlew build -PskipTests --no-daemon
+
+# [2] Runtime Stage
+FROM amazoncorretto:17 AS runtime
+WORKDIR /app
+
+# 빌드 결과물만 복사 (경로 주의!)
+COPY --from=build /app/build/libs/discodeit-1.2-M8.jar app.jar
+
+# 포트 노출 및 환경변수
+EXPOSE 80
+ENV JVM_OPTS=""
+
+# 앱 실행
+CMD ["sh", "-c", "java $JVM_OPTS -jar app.jar"]
```

sh -c 가 앞에 붙는다면 1번 프로세스는 쉘이 될겁니다. 아래처럼 바꿔주세요.

```suggestion
CMD ["sh", "-c", "exec", "java $JVM_OPTS -jar app.jar"]
```
해당 내용 참고해주세요. 자바 어플리케이션이 1번 프로세스가 되어야 합니다.
https://brunch.co.kr/@growthminder/142

shell form vs exec form 에 대해 알고 계셨으면 좋겠습니다.
그리고 왜 exec form을 써야 하는지도 알고 계셨으면 좋겠어요!
https://engineering.pipefy.com/2021/07/30/1-docker-bits-shell-vs-exec/